nodejs/eks/cluster.ts

```diff
@@ -560,8 +565,30 @@ ${customUserData}
 }
 
 async function computeWorkerSubnets(parent: pulumi.Resource, subnetIds: string[]): Promise<string[]> {
-    const subnets = await Promise.all(subnetIds.map(id => aws.ec2.getSubnet({id}, { parent: parent })));
-    const publicSubnets = subnets.filter(s => s.mapPublicIpOnLaunch);
-    const privateSubnets = subnets.filter(s => !s.mapPublicIpOnLaunch);
-    return (privateSubnets.length === 0 ? publicSubnets : privateSubnets).map(s => s.id);
+    const publicSubnets: string[] = [];
+    const privateSubnets: string[] = [];
+    for (const subnetId of subnetIds) {
+        const routeTable = await (async () => {
+            try  {
+                return await aws.ec2.getRouteTable({ subnetId: subnetId }, { parent: parent });
+            } catch {
```

Comment:
Under what condition does this throw?

Generally - comments on what this is trying to do and why it needs to do it this way would be helpful.

Reply:
> Generally - comments on what this is trying to do and why it needs to do it this way would be helpful.

Agreed. I will comment this up.